lib/secrets.rb

```diff
@@ -202,4 +261,12 @@ def read_vault_token(authfile_path)
   def with_retries(&block)
     Vault.with_retries(Vault::HTTPConnectionError, attempts: 3, &block)
   end
+
+  def params_from_path(path)
+    uri = URI.parse(path)
+    params = CGI.parse(uri.query).map do |key, value|
+      [key, value.length > 1 ? value : value[0]]
```

Comment:
describe what this logic does

Reply:
When requesting a signed cert from the PKI backend in Vault, you can optionally specify other properties of the cert you're requesting. For example, you can request a cert for a specific CN, or for a specific IP, etc. This `params_for_path` method takes arguments from the 'path' in the annotation (e.g. the `common_name=maxwell` portion of `pki/maxwell="pki/issue/maxwell?common_name=maxwell"`) and transforms those into the payload that's sent in the API request to Vault. 

agreed that the method name is vague, maybe renaming it to something like `pki_request_payload_from_path` would be more descriptive.

Reply:
yeah or put that comment into the method

Reply:
best put that as comment too so it's clear what this is used for ... or something like this as test description